src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1474,7 +1530,65 @@ protected override BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement n
                 }
             }
 
+            EnforceDoesNotReturn(node.Syntax);
             return null;
+
+            bool isUnconvertedBooleanExpression(BoundExpression expr)
+            {
+                var (expression, _) = RemoveConversion(expr, includeExplicitConversions: true);
+                return expression.Type?.SpecialType == SpecialType.System_Boolean;
```

Comment:
Can this be simplified to `expr.Kind != BoundKind.Conversion`?

Reply:
I worried that we might generate identity conversions.

---
In reply to: [370273577](https://github.com/dotnet/roslyn/pull/41098#discussion_r370273577) [](ancestors = 370273577)

Reply:
Thanks. This check was unnecessary after all. Removed.

---
In reply to: [370294344](https://github.com/dotnet/roslyn/pull/41098#discussion_r370294344) [](ancestors = 370294344,370273577)